### PR TITLE
Force HTTPS for selected domains

### PR DESCRIPTION
**Pull request by Alkarex**
Some feeds have embedded references using e.g. `//youtube.com/` or `http://youtube.com/` which are increasingly failing as HTTPS is required.

This PR allows setting a list of domains for which to force HTTPS with a new function `SimplePie::set_https_domains([])`

Downstream PR: https://github.com/FreshRSS/FreshRSS/pull/1087
https://github.com/FreshRSS/FreshRSS/pull/1098

### CODE REVIEW

File: library/SimplePie/Sanitize.php [hunk 1]
```diff
@@ -241,6 +249,68 @@ public function set_url_replacements($element_attribute = null)
 		$this->replace_url_attributes = (array) $element_attribute;
 	}
 
+	/**
+	 * Set the list of domains for which to force HTTPS.
+	 * @see SimplePie_Misc::https_url()
+	 * Example array('biz', 'example.com', 'example.org', 'www.example.net');
+	 */
+	public function set_https_domains($domains)
```
Comment (Alkarex) [inline library/SimplePie/Sanitize.php:257]: The list of domains is transformed into a tree for fast lookup and to avoid ambiguities.

**Comment by skyzyx** [comment]
Well I'm not opposed to this idea in general call mom I would hate to be on the hook for owning this list.

Is there anyway we could source a good list from somebody else who's already maintaining one? The EFF's HTTPS Everywhere extension comes to mind as a possible good source of data.

Assuming it's a JSON document, we could periodically fetch the list, transform it into a PHP lookup table, and ship a new release.￼￼￼

**Comment by skyzyx** [comment]
My mistake; I was on my phone and didn't see that this is a bring-your-own-list situation.

I think this is a good idea.

**Comment by Alkarex** [comment]
Not that it should be used, but just as an example, downstream, we use this list by default, customisable by end-users https://github.com/FreshRSS/FreshRSS/blob/edge/force-https.default.txt

Our two most important use-cases `dailymotion.com, youtube.com` are even set in the source-code at the moment but I will remove that if this PR is merged to keep the same code:

https://github.com/FreshRSS/FreshRSS/blob/634005de9a4b5e415ebf7c1106c769a0fbed5cfd/lib/SimplePie/SimplePie/Sanitize.php#L82

File: library/SimplePie/Sanitize.php [hunk 2]
```diff
@@ -441,7 +511,8 @@ public function replace_urls($document, $tag, $attributes)
 					if ($element->hasAttribute($attribute))
 					{
 						$value = $this->registry->call('Misc', 'absolutize_url', array($element->getAttribute($attribute), $this->base));
-						if ($value !== false)
+						$value = $this->https_url($value);
```
Comment (mblaney) [inline library/SimplePie/Sanitize.php:514]: `absolutize_url` can return false, I don't think you should check the url in that case.

Comment (Alkarex) [reply library/SimplePie/Sanitize.php]: Fixed https://github.com/simplepie/simplepie/pull/676/commits/5bfec781a947654b4f3610690be1572389e93033

**Pull request merged**